File: solutions/c/difference-of-squares/1/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int square_of_sum(unsigned int n){
    unsigned int i = 1;
    unsigned int sum = 0;
    while(i<=n){
        sum+=i;
        i++;
    }
    return sum*sum;
}

unsigned int sum_of_squares(unsigned int n){
    unsigned int i = 1;
    unsigned int sum = 0;
    while (i<=n){
        sum+=(i*i);
        i++;
    }
    return sum;
}

unsigned int difference_of_squares(unsigned int n){
    unsigned int n1 = square_of_sum(n);
    unsigned int n2 = sum_of_squares(n);
    return n1-n2;
}

File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	var i int = 1
    var s int = 0
    for i<=n {
        s+=i
        i++
    }
    return s*s
}

func SumOfSquares(n int) int {
	var i int = 1
    var s int = 0
    for i<=n {
        s+= i*i
        i++
    }
    return s
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
